src/node_file.cc

```diff
@@ -780,6 +780,32 @@ void AfterStat(uv_fs_t* req) {
   }
 }
 
+void OpenFileAfterOpen(uv_fs_t* open_req) {
+  FSReqBase* req_wrap = FSReqBase::from_req(open_req);
+  int result = static_cast<int>(open_req->result);
+
+  FS_ASYNC_TRACE_END1(
+      open_req->fs_type, req_wrap, "result", result)
+
+  if (result < 0) {
+    FSReqAfterScope after(req_wrap, open_req);
```

Comment:
This calls `uv_fs_req_cleanup()` here. Where is that function called if `result >= 0`, i.e. when this conditional is not hit?

Reply:
I probably misunderstood something, but as we pass `req_wrap->req()` to the `fstat` function, I thought that we should use the `FSReqAfterScope` only if we want to exit:
- In this case (when there is a problem with `open`
- In the `fstat` callback `AfterStat`

Am I correct?

Reply:
Yeah, `FSReqAfterScope` would destroy the `req_wrap` object; however, calling `uv_fs_req_cleanup()` is also an important part of what the function does, and if you don’t call it through `FSReqAfterScope`, you should call it directly after the libuv call.